### ISSUE TITLE
Raise an error for mapping method with non-instantiable target type

### ISSUE THREAD

**Issue by sjaakd** [mapstruct/mapstruct#319]
I noticed MapStruct tries to create a new abstract class in the following scenario:

``` java
public abstract class AbstractTarget() {
...
}

public interface SourceTargetMapper {

AbstractTarget mapFromSource(Source source);

}
```

Instead, MapStruct should advice to use the `@MappingTarget` construction.

**Comment by agudian** [comment]
Do we need this for 1.0.0.Final?

**Comment by sjaakd** [comment]
guess not.. The compiler does the same.. So nothing will break..

**Comment by gunnarmorling** [comment]
Yes, it's more the nice-to-have-category.

**Comment by sjaakd** [comment]
and relabelled accordingly  :smile:

**Comment by filiphr** [comment]
Isn't this the same as what we implemented in #1005?

**Comment by agudian** [comment]
Indeed it is! Let's close it...
